### ISSUE TITLE
Adds semicolon after the end of the code block

### ISSUE THREAD

**Issue by philipp-spiess** [krawaller/prettier-js-extension#2]
An additional semicolon after the last ``` block is inserted sometimes.

![](http://cloud.philippspiess.com/06080a283h28/Screen%20Recording%202017-03-22%20at%2006.45%20PM.gif)

**Comment by krawaller** [comment]
Good catch! It's because the naive way the formatting is applied – first it formats any eligible markdown code blocks, but after that it also tries to format the whole contents of the `<textarea>`, and it finds
```markdown
    ```js
    hello('world')
    ```
```
valid enough to format it as javascript – hence the extra semicolon 😆 

This can also be seen in the [prettier demo](https://prettier.github.io/prettier/#%7B%22content%22%3A%22%60%60%60js%5Cnhello('world')%5Cn%60%60%60%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22doc%22%3Afalse%7D%7D)

**Comment by krawaller** [comment]
We need a better heuristic to avoid false positives like this. Or maybe it shouldn't even try formatting the whole contents if it found code blocks to format? The naivety of the implementation is already starting to show 😬

**Comment by philipp-spiess** [comment]
Ha! Right, it's valid JS. 

> Or maybe it shouldn't even try formatting the whole contents if it found code blocks to format?

I think that would be good enough for now :)

**Comment by nickserv** [comment]
[Prettier 1.8 supports Markdown code blocks](https://prettier.io/blog/2017/11/07/1.8.0.html) and will properly format these examples by default if used in Markdown mode. We should update Prettier to at least 1.8 (preferably using a package manager or CDN) and configure Prettier to use Markdown when the text field is not JS.

**Comment by nickserv** [comment]
I started working on this and could not get Prettier to work. Unfortunately according to https://github.com/prettier/prettier/issues/3296, it has no browser support as of 1.7. They're making progress though, and I think it's worth waiting for, then I'll try to send a PR against a modern version of Prettier.
